### ISSUE TITLE
`Lint/UselessAssignment` false positive

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#9330]
The following should not raise an offense:

```ruby
x = true

module Foo
  if x
    def foo
    end
  end
end
```

```
$ rubocop -V
1.7.0 (using Parser 3.0.0.0, rubocop-ast 1.4.0, running on ruby 3.0.0 x86_64-darwin18)
```

**Comment by marcandre** [comment]
Not a false positive, my bad 🤦‍♂️
